src/components/UI/FileDrop/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { FileDropProps } from "@/types";
+import s from "./FileDrop.module.scss"
+import { useRef, useState } from "react";
+
+export const FileDrop = ({
+  children,
+  className,
+  onChange = (items) => { console.log('Changed Items: ', items) },
+  isMulti = false,
+}: FileDropProps) => {
+  const [isDraggerOver, setIsDraggedOver] = useState(false)
+  const fileInputRef = useRef<HTMLInputElement | null>(null)
+
+  const onChangeFiles = (files: FileList) => {
+    const filesNumber = files.length
+
+    if (!isMulti && filesNumber > 1) {
+      alert('Можливо додати лише один файл')
+      return;
+    }
+
+    onChange(files)
+  }
+
+  const handleDragEnter = (e: React.DragEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    setIsDraggedOver(true)
+  }
+
+  const handleDragOver = (e: React.DragEvent<HTMLDivElement>) => {
+    e.preventDefault();
```

Comment:
To avoid duplicating the word React in the types, you can import DragEvent and ChangeEvent from react.